genisys/main.py

```diff
@@ -27,17 +40,27 @@ def daemon():
     # TODO: Implement the daemon logic here
 
 def run(subcommand, args, module):
-    #netplan
-    netplan = net.Netplan(args)
+    # Config Parser
+    yamlParser = cp.YAMLParser()
+
+    # netplan
+    netplan = net.Netplan(args, yamlParser)
+
+    # preseed
+    preseed = ps.Preseed(args, yamlParser)
 
-    #preseed
-    preseed = ps.Preseed(args)
+    # nat
+    nat = nt.Nat(args, yamlParser)
 
-    #nat
-    nat = nt.Nat(args)
+    # kernelparameter
+    kernelParameter = kp.KernelParameter(args, yamlParser)
 
-    #kernelparameter
-    kernelParameter = kp.KernelParameter(args)
+    # setup commands
+    modulesList = [netplan, preseed, nat, kernelParameter]
+    for mod in modulesList:
```

Comment:
As is, this is gonna run our setup commands every time the application is ran. We should move this to inside of the `subcommand == "install"` branch of the `if` statement below. I also see we don't have a case for when the subcommand is generate. Right now, the difference between "install" and "generate" is that this loop of commands should run for "install" but not "generate". Other wise, "generate" should work the same.